baseframe/forms/validators.py

```diff
@@ -16,23 +16,38 @@
 from ..signals import exception_catchall
 
 
-__all__ = ['OptionalIf', 'OptionalIfNot', 'ValidEmail', 'ValidEmailDomain', 'ValidUrl', 'AllUrlsValid',
-    'ValidName', 'NoObfuscatedEmail', 'ValidCoordinates', 'IsPublicEmailDomain', 'IsNotPublicEmailDomain',
-    # WTForms validators
-    'DataRequired', 'InputRequired', 'Optional', 'Length', 'EqualTo', 'URL', 'NumberRange',
-    'ValidationError', 'StopValidation']
+__local = ['AllUrlsValid', 'IsNotPublicEmailDomain', 'IsPublicEmailDomain', 'NoObfuscatedEmail',
+    'OptionalIf', 'OptionalIfNot', 'RequiredIf', 'ValidCoordinates', 'ValidEmail',
+    'ValidEmailDomain', 'ValidName', 'ValidUrl']
+__imported = [  # WTForms validators
+    'DataRequired', 'EqualTo', 'InputRequired', 'Length', 'NumberRange', 'Optional',
+    'StopValidation', 'URL', 'ValidationError']
+__all__ = __local + __imported
 
 
 EMAIL_RE = re.compile(r'\b[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,63}\b', re.I)
 
 
+class AllowedIf(object):
+    """
+    Validator that allows a value only if some other field has a value.
+    """
+    def __init__(self, fieldname, message=None):
+        self.fieldname = fieldname
+        self.message = message or __("This field is not allowed")
+
+    def __call__(self, form, field):
+        if field.data and not form[self.fieldname].data:
+            raise StopValidation(self.message)
+
+
 class OptionalIf(object):
     """
     Validator that makes this field optional if the value of some other field is true.
     """
     def __init__(self, fieldname, message=None):
         self.fieldname = fieldname
-        self.message = message or __("This is required")
+        self.message = message or __("This field is required")
```

Comment:
"field" is not something users understand. The earlier message is more appropriate.